core/src/io/anuke/mindustry/type/Category.java

```diff
@@ -23,4 +25,14 @@
     effect;
 
     public static final Category[] all = values();
+
+    Block selectedBlock = null;
```

Comment:
1. This should not be stored in the category, as it will lead to very interesting bugs if a game reloads on Android or after a mod. Use an `ObjectMap` in input instead that gets cleared on every `ResetEvent`.
2. Don't use getters and setters.

Reply:
1. Done in https://github.com/Anuken/Mindustry/pull/837/commits/6ea949290e95005e50b652abc2ff7d23d25c7bca
2. For my personnal information, why shouldn't getters/setters be used in this case? I didn't do Java since university and back then we were taught that getters and setters are a must 😅

Reply:
Getters and setters bloat the code and aren't really necessary unless you're making a library. If you need to encapsulate the variable at a later point in time, IntelliJ can do that for you in a few clicks.